### ISSUE TITLE
LLVM `If`/`Else` basic blocks are misbehaving

### ISSUE THREAD

**Issue by Champii** [Champii/Rock#42]
The codegen part is buggy when it comes to final basic block and terminal value collecting.

**Comment by Champii** [comment]
Investigate the use of PHI values

**Comment by Champii** [comment]
Minimal reproducible snippet

```haskell
main =
    if true
    then 1
    else 2
    0
```

```
Error: Bug in the generated IR:

Basic Block in function 'main' does not have terminator!
label %then
```

**Comment by Champii** [comment]
Fixed in 901660014eb9ea5c5ab750a850d3a6204e91480e
